addons/point_of_sale/tests/test_frontend.py

```diff
@@ -1204,6 +1206,30 @@ def test_res_partner_scan_barcode(self):
         self.main_pos_config.with_user(self.pos_user).open_ui()
         self.start_tour("/pos/ui?config_id=%d" % self.main_pos_config.id, 'BarcodeScanPartnerTour', login="pos_user")
 
+    def test_allow_order_modification_after_validation_error(self):
+        """
+        User error as a result of validation should block the order.
+        Taking action by order modification should be allowed.
+        """
+
+        self.env['product.product'].create({
+            'name': 'Test Product',
+            'list_price': 10.00,
+            'taxes_id': False,
+            'available_in_pos': True,
+        })
+
+        def sync_from_ui_patch(*_args, **_kwargs):
+            raise UserError('Test Error')
+
+        with patch.object(type(self.env['pos.order']), "sync_from_ui", sync_from_ui_patch):
```

Comment:
Thank you for the suggestion. However, if I applied the suggestion and remove the `assertLogs` context, I get the following log:

![Screenshot 2024-04-25 at 10 56 38](https://github.com/odoo/odoo/assets/3245568/d6406945-554d-447e-acdf-c55536eb23b6)

I actually expect the following log:

![Screenshot 2024-04-25 at 10 57 27](https://github.com/odoo/odoo/assets/3245568/249da5af-319a-4e97-a3e4-957c2a00ce2e)

I added a commit to specify the type of log I'm expecting. I'll squash when you're okay with the changes.

Reply:
> Get the class from the registry instead of deriving it from an instance.

I just saw that comment. I'm not sure if I understood it correctly, but after some debugging and doing `type(self.env['pos.order'])` is just giving the same result as `self.env.registry.models['pos.order']`. Do I need to replace my use of `type(self.env['pos.order'])`?

Reply:
> Do I need to replace my use of `type`?

Yes please

Reply:
I applied the following change to avoid the use of type and squash the commits.

<img width="965" alt="Screenshot 2024-04-29 at 09 37 44" src="https://github.com/odoo/odoo/assets/3245568/7d9ecb35-f847-44cb-ba49-48fe777497dc">